File: python/Binary Tree Level Order Traversal II.py
def levelOrderBottom(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if root is None:
            return []
        result = []
        que = [root]
        
        while len(que) != 0:
            level = []
            size = len(que)
            
            for i in range(size):
                node = que.pop(0)
                level.append(node.val)
                if node.left:
                    que.append(node.left)
                if node.right:
                    que.append(node.right)
            result.insert(0,level)
        return result

File: java/Binary Tree Level Order Traversal II.java
List<List<Integer>> result = new ArrayList<List<Integer>>();
        
        Queue<TreeNode> que = new LinkedList<TreeNode>();
        
        if(root == null) return result;
        que.offer(root);
        
        while(!que.isEmpty()) {
            
            ArrayList<Integer> level = new ArrayList<Integer>();
            int size = que.size();
            
            for(int i = 0; i < size;i++) {
                TreeNode node = que.poll();
                level.add(node.val);
                
                if(node.left != null){
                    que.offer(node.left);
                }
                if(node.right != null){
                    que.offer(node.right);
                }
            }
            
            result.add(0,level);
            
        }
        
        return result;
